test/Driver/EDMF/dycore_kernels.jl

```diff
@@ -0,0 +1,285 @@
+)
+    nodal_update_auxiliary_state!(single_stack_nodal_update_aux!, dg, m, Q, t, elems)
+    update_auxiliary_state!(dg, m, m.edmf, Q, t, elems)
+end;
+
+# Compute/update all auxiliary variables at each node. Note that
+# - `aux.T` is available here because we've specified `T` in
+# `vars_state_auxiliary`
+function single_stack_nodal_update_aux!(
+    m::SingleStack{FT,N},
+    state::Vars,
+    aux::Vars,
+    t::Real,
+) where {FT, N}
+    aux.buoyancy = 0
+end;
+
+# Since we have second-order fluxes, we must tell `ClimateMachine` to compute
+# the gradient of `ρcT`. Here, we specify how `ρcT` is computed. Note that
+#  - `transform.ρcT` is available here because we've specified `ρcT` in
+#  `vars_state_gradient`
+function compute_gradient_argument!(
+    m::SingleStack{FT,N},
+    transform::Vars,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+) where {FT, N}
+    compute_gradient_argument!(m, m.edmf, state, aux, t)
+end;
+
+# Specify where in `diffusive::Vars` to store the computed gradient from
+# `compute_gradient_argument!`. Note that:
+#  - `diffusive.α∇ρcT` is available here because we've specified `α∇ρcT` in
+#  `vars_state_gradient_flux`
+#  - `∇transform.ρcT` is available here because we've specified `ρcT`  in
+#  `vars_state_gradient`
+function compute_gradient_flux!(
+    m::SingleStack{FT,N},
+    diffusive::Vars,
+    ∇transform::Grad,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+) where {FT,N}
+    compute_gradient_flux!(m, m.edmf, diffusive, ∇transform, state, aux, t)
+end;
+
+# We have no sources, nor non-diffusive fluxes.
+function source!(
+    m::SingleStack{FT, N},
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    direction,
+) where {FT, N}
+    source!(m, m.edmf, state, diffusive, aux, t, direction)
+end;
+
+function flux_first_order!(
+    m::SingleStack,
+    flux::Grad,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+
+    ρinv = 1/state.ρ
+    flux.ρ = state.ρu
+    u = state.ρu * ρinv
+    flux.ρu = state.ρu * u'
+    flux.ρe_int = u * state.ρe_int
+    flux.ρq_tot = u * state.ρq_tot
+    flux_first_order!(m, m.edmf, flux, state, aux, t)
+end;
+
+function flux_second_order!(
+    m::SingleStack{FT,N},
+    flux::Grad,
+    state::Vars,
+    diffusive::Vars,
+    hyperdiffusive::Vars,
+    aux::Vars,
+    t::Real,
+) where {FT,N}
+    # TODO: This should be in sync with `flux_second_order!(::AtmosModel)`, plus
+    # the call to `flux_second_order!(m, m.edmf, flux, state, diffusive, hyperdiffusive, aux, t)`
+
+    # Aliases:
+    up   = state.edmf.updraft
+    en   = state.edmf.environment
+    en_d = diffusive.edmf.environment
+
+    # compute the mixing length and eddy diffusivity
+    # (I am repeating this after doing in sources assuming that it is better to compute this twice than to add mixing length as a aux.variable)
+    l = mixing_length(m, m.edmf.mix_len, source, state, diffusive, aux, t, direction, δ, εt)
+    K_eddy = m.c_k*l*sqrt(en.tke)
```

Comment:
We need to reconcile the differences between this kernel and what’s in the `AtmosModel` kernel